core/trino-parser/src/main/java/io/trino/sql/tree/CreateTable.java

```diff
@@ -27,25 +27,27 @@
 {
     private final QualifiedName name;
     private final List<TableElement> elements;
+    private final boolean replace;
     private final boolean notExists;
     private final List<Property> properties;
     private final Optional<String> comment;
 
-    public CreateTable(QualifiedName name, List<TableElement> elements, boolean notExists, List<Property> properties, Optional<String> comment)
+    public CreateTable(QualifiedName name, List<TableElement> elements, boolean replace, boolean notExists, List<Property> properties, Optional<String> comment)
```

Comment:
Having so many boolean arguments is error prone -- it's hard for a reader to see what's what when looking at a callsite.

Also, replace and notExists should be exclusive of each other. It's not a valid combination to do "CREATE OR REPLACE ... IF NOT EXISTS". See https://github.com/trinodb/trino/issues/13180#issuecomment-1259698248

Reply:
Fair enough, let's go with an enum of all supported combinations.

Reply:
>Having so many boolean arguments is error prone

What do you suggest? If we made them mutually exclusive in the grammar, we could have an enum to represent the options when table exists: fail, replace, or ignore.

Reply:
In the grammar would give an error as follows:

```
assertStatementIsInvalid("CREATE OR REPLACE TABLE IF NOT EXISTS foo AS SELECT * FROM t")
                .withMessage("line 1:43: mismatched input 'AS'. Expecting: '(', '.'");
```

Let me know if you think that's better than throwing a more specific exception.

Reply:
Note that materialized views also have similar behavior.  2 boolean flags for `replace` and `notExists` and a explicit error in `StatementAnalyzer` (`throw semanticException(NOT_SUPPORTED, node, "'CREATE OR REPLACE' and 'IF NOT EXISTS' clauses can not be used together");`).

Reply:
If we won't allow a instance (CreateTable or CreateMaterializedView) to be created, we will have to throw the semanticException in `AstBuilder`, this doesn't seem consistent with the existing code.

Reply:
> In the grammar would give an error as follows: ...

@mdesmet I think this is an important concern.
If we can't get a meaningful message from the parser error handler, there's another way. See `AstBuilder.visitTrim()` for reference. The idea is that we allow all combinations in the grammar (like in the previous version of the grammar, which btw looked much clearer). Then we catch the invalid combinations in the `AstBuilder`, and throw a parsing exception by hand.

Reply:
I have reverted back to the old version of the grammar and throw a `parseError` instead. Thanks for the guidance!